addons/purchase/models/purchase.py

```diff
@@ -1220,6 +1220,8 @@ def _compute_price_unit_and_date_planned_and_name(self):
         for line in self:
             if not line.product_id or line.invoice_lines:
                 continue
+            if not line.order_id.date_order:
+                line.order_id.date_order = fields.Date.context_today(line)
```

Comment:
It's usually not a good idea to set values in `compute` method for fields that are not intended to be. 
In this case you can give `today` directly to `_select_seller` (and any next call needing a date)

Reply:
@omra-odoo